### PR TITLE
Mediaklikk extractor

### PR DESCRIPTION
**Pull request by tmarki**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Extractor for Hungarian state media (MTVA) video repository hosted at https://www.mediaklikk.hu/mediatar/

### CODE REVIEW

File: youtube_dl/extractor/mediaklikk.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                'title': 'Sporthíradó, 350. rész',
+                'series': 'Sporthíradó',
+                'ext': 'mp4'
+            }
+        },
+    ]
+
+    def _real_extract(self, url):
+        webpage = self._download_webpage(url,
+                                         None,
```
Comment (dstftw) [inline youtube_dl/extractor/mediaklikk.py]: Capture and output display id.

File: youtube_dl/extractor/mediaklikk.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                                         note='Fetching page')
```
Comment (dstftw) [inline youtube_dl/extractor/mediaklikk.py]: Default note is enough.

File: youtube_dl/extractor/mediaklikk.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        pattern = r"mtva_player_manager\.player\(document.getElementById\(.*\),(.*)\);"
```
Comment (dstftw) [inline youtube_dl/extractor/mediaklikk.py]: Capturing empty string is senseless. If you expect a dict then capture group must contain `{}`.

File: youtube_dl/extractor/mediaklikk.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        info_json = self._html_search_regex(pattern, webpage, 'info_json')
+        info_meta = self._parse_json(compat_urllib_parse_unquote(info_json),
+                                     None)
+        video_id = str(info_meta['contentId'])
```
Comment (dstftw) [inline youtube_dl/extractor/mediaklikk.py]: compat_str

File: youtube_dl/extractor/mediaklikk.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            'title': info_meta.get('title') or self._og_search_title(webpage),
+            'ext': 'mp4',
+            'id': video_id
+        }
+        if 'series' in info_meta:
+            info_ret['series'] = info_meta['series']
+        info_meta['video'] = info_meta['token']
+        del info_meta['token']
+        playerpage = self._download_webpage('https://player.mediaklikk.hu/playernew/player.php', video_id, note='Downloading player page', query=info_meta)
+        pattern = r"\"file\": \"(.*)\","
```
Comment (dstftw) [inline youtube_dl/extractor/mediaklikk.py]: Same.

**Comment by tmarki** [comment]
Please see commit with fixes.

**Comment by dstftw** [comment]
Also read coding conventions.

**Comment by mrx23dot** [comment]
Let's not hang up on nuances

**Comment by mrx23dot** [comment]
Finished work in #25098

**Comment by VinhoXie** [comment]
Work well. You can also have a check at this article about [mediaklikk downloader](https://enjoy-online-audio-video-movie.blogspot.com/2021/05/top-3-downloaders-for-mediaklikk.html).

**Pull request closed without merging**